specification/cognitiveservices/data-plane/WebSearch/v1.0/WebSearch.json

```diff
@@ -0,0 +1,1211 @@
+{
+  "swagger": "2.0",
+  "info": {
+    "title": "Web Search API",
+    "description": "The Web Search API lets you send a search query to Bing and get back search results that include links to webpages, images, and more.",
+    "version": "1.0"
+  },
+  "host": "api.cognitive.microsoft.com",
+  "schemes": [
+    "https"
+  ],
+  "basePath": "/bing/v7.0",
+  "produces": [
+    "application/json"
+  ],
+  "securityDefinitions": {
+    "apiKeyHeader": {
+      "name": "Ocp-Apim-Subscription-Key",
+      "type": "apiKey",
+      "in": "header"
+    },
+    "apiKeyQuery": {
+      "name": "subscription-key",
+      "type": "apiKey",
+      "in": "query"
+    }
+  },
+  "security": [
+    {
+      "apiKeyHeader": []
+    },
+    {
+      "apiKeyQuery": []
+    }
+  ],
+  "paths": {
+    "/search": {
+      "get": {
+        "summary": "The Web Search API lets you send a search query to Bing and get back search results that include links to webpages, images, and more.",
+        "operationId": "Web_Search",
+        "tags": [
+          "WebSearch"
+        ],
+        "parameters": [
+          { "$ref": "#/parameters/x-bingapis-sdk" },
```

Comment:
This is one heckova API.

With all the parameters, this makes a hell of a method:

``` csharp
public SearchResponse Search(
   string query, 
   string acceptLanguage = default(string), 
   string pragma = default(string), 
   string userAgent = default(string), 
   string clientId = default(string), 
   string clientIp = default(string), 
   string location = default(string), 
   string answerCount = default(string), 
   string countryCode = default(string), 
   string count = default(string), 
   string freshness = default(string), 
   string market = default(string), 
   string offset = default(string), 
   string promote = default(string), 
   IList<AnswerType?> responseFilter = default(IList<AnswerType?>), 
   SafeSearch? safeSearch = default(SafeSearch?), 
   string setLang = default(string), 
   string textDecorations = default(string), 
   string textFormat = default(string))
```

I wonder if we can move some of the parameters down to the global section and make them client parameters instead of method parameters. That would go a long way to simplifying the method definition.

Reply:
Does that mean those parameters would be set once when the client is instantiated? Maybe a few can, but most are definitely volatile per request

Reply:
Oh, I get it, there is a lot that you'd want per-request, but if some can be moved up it'll help a bit.

The parameters in the client are still mutable properties, so they can be changed.

Reply:
If the client is meant to be thread safe (I can only hope that it is 😜 ) this gets into hairy territory.

Do you mind if we accept the ugliness of the parameters for now, and we can take an item as a team to decide what makes the most sense for users of the SDK? We have the same issue for entity search (and we will for the other Bing APIs), someone actually filed an issue on it: https://github.com/Azure/azure-rest-api-specs/issues/1728 suggesting parameter groupings as an option.

Reply:
The only tricky bit is that moving a parameter to global later will be a binary breaking change in the API, and you'll want to bump the version.

Now that being said, I'm no opposed to that either. I think a rapidly moving API version makes people excited 🥇